src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_copy_b.cpp

```diff
@@ -107,8 +108,42 @@ std::shared_ptr<Node> intel_cpu::BrgemmCopyB::clone_with_new_inputs(const Output
                                          m_K_blk, m_N_blk);
 }
 
-size_t intel_cpu::BrgemmCopyB::get_offset_compensations() const {
+size_t BrgemmCopyB::get_offset_compensations() const {
     OPENVINO_ASSERT(is_with_compensations() && get_output_size() == 2,
                     "The offset for compensations must be in BrgemmCopyB only with compensations and 2 outputs!");
     return get_output_offset(1);
 }
+
+BrgemmCopyB::ShapeInfer::ShapeInfer(const std::shared_ptr<ov::Node>& n) {
+    const auto& brg_copyb = ov::as_type_ptr<BrgemmCopyB>(n);
+    OPENVINO_ASSERT(brg_copyb, "Got invalid node in BrgemmCopyB::ShapeInfer");
+    m_layout = snippets::lowered::PortDescriptorUtils::get_port_descriptor_ptr(n->input(0))->get_layout();
+    m_num_outs = brg_copyb->is_with_compensations() ? 2 : 1;;
+    m_N_blk = brg_copyb->get_input_element_type(0) == element::bf16 ? 32 : 64;
```

Comment:
Is it possible to take `m_N_blk` from `brg_copyb` node?

Reply:
It is, sure. I also moved `brgemmVNNIFactor` to a dedicated op field, since ideally `ShapeInfer` shouldn't contain any logic that calculates any of the node's parameters (other than out shapes).